health-checker/src/main/java/io/stargate/health/Server.java

```diff
@@ -70,6 +75,21 @@ protected void configure() {
     CollectorRegistry.defaultRegistry.register(new DropwizardExports(metrics.getRegistry()));
   }
 
+  private void maybeBindToListenAddressOnly(ApplicationConfiguration applicationConfiguration) {
+    boolean bindToListenAddressOnly = Boolean.getBoolean("stargate.bind_to_listen_address");
+    if (bindToListenAddressOnly) {
+      DefaultServerFactory server =
+          (DefaultServerFactory) applicationConfiguration.getServerFactory();
+      ConnectorFactory connector = server.getAdminConnectors().get(0);
+      if (connector instanceof HttpConnectorFactory) {
+        ((HttpConnectorFactory) connector)
+            .setBindHost(System.getProperty("stargate.listen_address"));
+      } else {
+        throw new AssertionError("Unexpected connector class: " + connector.getClass().getName());
+      }
+    }
```

Comment:
maybe add `log.debug` in the else branch to not "fail" silently?

Reply:
The exception should bubble up and appear in the Stargate logs I think.